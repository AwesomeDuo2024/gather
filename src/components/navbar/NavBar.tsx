"use client";

import * as React from "react";
import Link from "next/link";
import Image from "next/image";

import {
  NavigationMenu,
  // NavigationMenuContent,
  NavigationMenuItem,
  NavigationMenuLink,
  NavigationMenuList,
  // NavigationMenuTrigger,
  navigationMenuTriggerStyle,
} from "@/components/ui/navigation-menu";

export default function NavBar() {
  return (
    <div className="w-screen bg-zinc-950">
      <div className="container flex justify-between items-center px-5 md:px-20 py-5">
        <div className="hover:cursor-pointer">
          <Link href="/" legacyBehavior passHref>
            <Image
              src="/logo.svg"
              width={100}
              height={100}
              alt="Gather logo"
              className="w-auto"
            />
          </Link>
        </div>
        <NavigationMenu>
          <NavigationMenuList>
            <NavigationMenuItem className="flex gap-8">
              <Link href="/#faq" legacyBehavior passHref>
                <NavigationMenuLink className="text-gray-500 hover:text-gray-200 transition-all text-sm lg:text-base">
                  How to use
                </NavigationMenuLink>
              </Link>
              <Link
                href="https://donate.stripe.com/test_00g7uBbfkcjebNm6oo"
                legacyBehavior
                passHref
              >
                <NavigationMenuLink className="text-gray-500 hover:text-gray-200 transition-all text-sm lg:text-base">
                  Donate
                </NavigationMenuLink>
              </Link>
            </NavigationMenuItem>
          </NavigationMenuList>
        </NavigationMenu>
      </div>
    </div>
  );
}
