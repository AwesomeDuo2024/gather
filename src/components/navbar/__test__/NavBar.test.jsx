// https://refine.dev/blog/react-unit-testing/#snapshot-testing
// https://nextjs.org/docs/app/building-your-application/testing/jest

import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import NavBar from "../NavBar";
import React from "react";

describe("NavBar", () => {
  test("should render logo", () => {
    // render component in virtual dom
    render(<NavBar />);

    // select elements
    const logo = screen.getByTestId("logo");

    // assert presence and/or text of elements selected
    expect(logo).toBeInTheDocument();
  });

  test("should render 'How to use' nav link", () => {
    render(<NavBar />);
    const faqLink = screen.getByTestId("faq-link");
    expect(faqLink).toBeInTheDocument();
    expect(faqLink).toHaveTextContent("How to use");
  });

  test("should render 'Donate' nav link", () => {
    render(<NavBar />);
    const donateLink = screen.getByTestId("donate-link");
    expect(donateLink).toBeInTheDocument();
    expect(donateLink).toHaveTextContent("Donate");
  });
});

describe("NavBar component matches snapshot", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<NavBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
