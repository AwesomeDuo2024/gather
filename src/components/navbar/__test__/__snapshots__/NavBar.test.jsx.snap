// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component matches snapshot should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-screen bg-zinc-950"
  >
    <div
      class="container flex justify-between items-center px-5 md:px-20 py-5"
    >
      <div
        class="hover:cursor-pointer"
      >
        <img
          alt="Gather logo"
          data-nimg="1"
          data-testid="logo"
          decoding="async"
          height="110"
          href="/"
          loading="lazy"
          src="/logo.svg"
          style="color: transparent;"
          width="110"
        />
      </div>
      <nav
        aria-label="Main"
        class="relative z-10 flex max-w-max flex-1 items-center justify-center"
        data-orientation="horizontal"
        dir="ltr"
      >
        <div
          style="position: relative;"
        >
          <ul
            class="group flex flex-1 list-none items-center justify-center space-x-1"
            data-orientation="horizontal"
            dir="ltr"
          >
            <li
              class="flex gap-8"
            >
              <a
                class="text-gray-500 hover:text-gray-200 transition-all text-sm lg:text-base"
                data-radix-collection-item=""
                data-testid="faq-link"
                href="/#faq"
              >
                How to use
              </a>
              <a
                class="text-gray-500 hover:text-gray-200 transition-all text-sm lg:text-base"
                data-radix-collection-item=""
                data-testid="donate-link"
                href="https://donate.stripe.com/test_00g7uBbfkcjebNm6oo"
              >
                Donate
              </a>
            </li>
          </ul>
        </div>
        <div
          class="absolute left-0 top-full flex justify-center"
        />
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
